### ISSUE TITLE
220-rc1 manager won't load

### ISSUE THREAD

**Issue by goldsky** [modxbot/migrate#6063]
goldsky created Redmine issue ID 6063

I just played with 220-rc1 and I got this after a successful login.
It applied to both traditional and basic distributions.

**Comment by goldsky** [comment]
goldsky submitted:

It applied to both traditional and advanced distributions.

**Comment by opengeek** [comment]
opengeek submitted:

Are there any error logs with relevant information? Did you make sure and clear your browser caches? I can't reproduce this behavior. Anyone else?

**Comment by goldsky** [comment]
goldsky submitted:

I bet there is a problem with my server setting, or something change on the revo's basic requirements.
Here is my complete server's info.

**Comment by splittingred** [comment]
splittingred submitted:

Any errors in core/cache/logs/error.log?

Also, check out the following settings:
- manager_js_cache_file_locking - Set to No if filesystem is NFS.
- manager_js_cache_max_files - Set to something like 5, since I bet minify is choking on your slow env.

**Comment by goldsky** [comment]
goldsky submitted:

Shaun McCormick wrote:

> Any errors in core/cache/logs/error.log?

No, nothing.

Shaun McCormick wrote:

> Also, check out the following settings:
> 
> > - manager_js_cache_file_locking - Set to No if filesystem is NFS.
> > - manager_js_cache_max_files - Set to something like 5, since I bet minify is choking on your slow env.

Manually?
remember, I can't get into the system settings.

**Comment by wshawn** [comment]
wshawn submitted:

Just curious: did you try another browser or empty the cache on the one you have been using? Even though this was reportedly fixed, I recently experienced the same issue.

**Comment by skywaylord** [comment]
skywaylord submitted:

Are there any Aliases, Symlinks or Virtual Directories in your server environment? It seems to be a minify related problem:

http://code.google.com/p/minify/wiki/UriRewriting#Aliases_/_Symlinks_/_Virtual_Directories

If so, you need to change $min_symlinks in \manager\min\index.php on line 54.

**Comment by goldsky** [comment]
goldsky submitted:

W ShawnWilkerson wrote:

> Just curious: did you try another browser or empty the cache on the one you have been using? Even though this was reportedly fixed, I recently experienced the same issue.

This is definitely not browser specific.
Have tried on FF, Opera, Chrom, Safari, all looked the same.

SWL Team wrote:

> Are there any Aliases, Symlinks or Virtual Directories in your server environment? It seems to be a minify related problem:
> 
> http://code.google.com/p/minify/wiki/UriRewriting#Aliases_/_Symlinks_/_Virtual_Directories
> 
> If so, you need to change $min_symlinks in \manager\min\index.php on line 54.

I haven't done anything yet.
The manager broke from the beginning.

**Comment by opengeek** [comment]
opengeek submitted:

He is asking about server environment, not MODX, with regards to Aliases, Symlinks, and Virtual Directories.

**Comment by frippz** [comment]
frippz submitted:

I recognize this. I think I got the same problem when I used minify on my host. They happen to lack the  $_SERVER['DOCUMENT_ROOT'] variable and I needed to configure minify accordingly. See http://code.google.com/p/minify/source/browse/trunk/min/config.php?r=292#47

Hope this helps a little.

**Comment by frippz** [comment]
frippz submitted:

I can now confirm that setting $min_documentRoot on line 54 in /manager/min/index.php manually works. That is the same solution I used to fix my own minify problem. I'm not really sure how to get that value dynamically when $_SERVER['DOCUMENT_ROOT'] is not set (which seems to be not too uncommon on some hosts). But I guess someone more skilled than me knows how. :)

**Comment by monnie** [comment]
monnie submitted:

I'm having the same issue after installing Modx 2.2.0-rc1 (from the pre-release) on my local host. I get through the installation process just fine, but when I log into the manager I get the same page as seen in 220-rc1-manager-error.jpg above. In the console in firebug, I get several 400 Bad Request errors. The CSS is breaking and there is definitely a conflict with minify. I have tried manually setting $min_documentRoot and $min_symlinks in /manager/min/index.php. I'm using Apache on a Mac. I do not fully understand Aliases or Symlinks. Can I get an example of what I should set the $min_documentRoot to? I've read a bit of documentation on it, and I imagine it is something like /Users/accountname/Sites or //localhost/~accountname but I can't get it to work. Also, can I set it after the installation, or does it have to be set before running setup?

Thank you for your input so far on this issue. It has helped me pinpoint the problem, but not quite resolve it.

**Comment by splittingred** [comment]
splittingred submitted:

Fixed here: https://github.com/modxcms/revolution/commit/6baa569956a4826b4948fe2f8b8f00c60edc27de

There is now a new setting, "manager_js_document_root", that can be set in System Settings to explicitly set the document root for minify. Also, I added a check to force the $min_documentRoot setting to the calculated $_SERVER['DOCUMENT_ROOT'] at the beginning of the min/index.php file. 

Finally, I added a check in setup/ that forces CSS/JS compression off if the server DOCUMENT_ROOT variable is empty. These fixes should deal with any issues that occur.
